src/legacy/core_plugins/console/server/proxy_route.js

```diff
@@ -105,42 +103,54 @@ export const createProxyRoute = ({
       const { path, method } = query;
       const uri = resolveUri(baseUrl, path);
 
+      // Because this can technically be provided by a settings-defined proxy config, we need to
+      // preserve these property names to maintain BWC.
       const { timeout, rejectUnauthorized, agent, headers } = getConfigForReq(req, uri);
-      const makeRequest = async payloadToSend => {
-        const wreckOptions = {
-          payload: payloadToSend,
-          timeout,
-          rejectUnauthorized,
-          agent,
-          headers: {
-            ...headers,
-            ...getProxyHeaders(req),
-          },
-        };
-
-        const esResponse = await Wreck.request(method, uri, wreckOptions);
 
-        if (method.toUpperCase() !== 'HEAD') {
-          return h
-            .response(esResponse)
-            .code(esResponse.statusCode)
-            .header('warning', esResponse.headers.warning);
-        }
+      return await new Promise((resolve) => {
+        const pathParts = path.split('?'); // TODO: Ensure this only has 2 parts
+        const requestHeaders = {
+          ...headers,
+          ...getProxyHeaders(req),
+        };
 
-        return h
-          .response(`${esResponse.statusCode} - ${esResponse.statusMessage}`)
-          .code(esResponse.statusCode)
-          .type('text/plain')
-          .header('warning', esResponse.headers.warning);
-      };
-      // Wreck assumes that DELETE requests will not have a body, and thus it does not
-      // parse the payload to pass it along, so we have to do this manually here.
-      if (method.toUpperCase() === 'DELETE') {
-        const data = await Wreck.read(payload);
-        return await makeRequest(data);
-      } else {
-        return await makeRequest(payload);
-      }
+        // We use the request library because Hapi, Axios, and Superagent don't support GET requests
+        // with bodies, but ES APIs do. Similarly with DELETE requests with bodiese. If we need to
+        // deprecate use of this library, we can also solve this issue with Node's http library.
+        // See #39170 for details.
+        request({
+          method,
+          uri,
+          timeout,
+          qs: parse(pathParts[1] || ''),
+          headers: requestHeaders,
+          body: payload ? JSON.stringify(payload) : undefined,
+          // agentOptions,
```

Comment:
[agentOptions](https://github.com/request/request#using-optionsagentoptions) can be used to pass along SSL protocol options, though the docs recommend passing them along directly on the `options` object.